Sources/joyfill-components-swift/Components/Table/ViewTable.swift

```diff
@@ -457,41 +554,46 @@ public class ViewTable: UIViewController, TextFieldCellDelegate, TextViewCellDel
     @objc func moveUpTapped() {
         updateRowNumber()
         if cellSelectedIndexPath?.section == 1 {
-            
+            updateSelectionButton()
+            selectedIndexPath = nil
+            updateCellSubviewBorder(at: 0)
         } else {
             updateSelectionButton()
             collectionView.moveSection(cellSelectedIndexPath?.section ?? 0, toSection: (cellSelectedIndexPath?.section ?? 0) - 1)
-            textViewTextArray.swapAt(cellSelectedIndexPath?.section ?? 0, (cellSelectedIndexPath?.section ?? 0) - 1)
+            valueData.swapAt((cellSelectedIndexPath?.section ?? 0) - 1, (cellSelectedIndexPath?.section ?? 0) - 2)
+            tableCellsData.swapAt((cellSelectedIndexPath?.section ?? 0) - 1, (cellSelectedIndexPath?.section ?? 0) - 2)
+            selectedIndexPath = nil
+            updateCellSubviewBorder(at: -1)
         }
-        selectedIndexPath = nil
-        updateCellSubviewBorder(at: -1)
     }
     
     // MARK: MoveDown Tapped
     @objc func moveDownTapped() {
         updateRowNumber()
-        if cellSelectedIndexPath?.section == 0 || cellSelectedIndexPath?.section == numberOfRows - 1 {
-            
+        if cellSelectedIndexPath?.section == 0 || cellSelectedIndexPath?.section == numberOfRows {
+            updateSelectionButton()
+            selectedIndexPath = nil
+            updateCellSubviewBorder(at: 0)
         } else {
             updateSelectionButton()
             collectionView.moveSection(cellSelectedIndexPath?.section ?? 0, toSection: (cellSelectedIndexPath?.section ?? 0) + 1)
-            textViewTextArray.swapAt(cellSelectedIndexPath?.section ?? 0, (cellSelectedIndexPath?.section ?? 0) + 1)
+            valueData.swapAt((cellSelectedIndexPath?.section ?? 0) - 1, cellSelectedIndexPath?.section ?? 0)
+            tableCellsData.swapAt((cellSelectedIndexPath?.section ?? 0) - 1, cellSelectedIndexPath?.section ?? 0)
+            selectedIndexPath = nil
+            updateCellSubviewBorder(at: 1)
         }
-        selectedIndexPath = nil
-        updateCellSubviewBorder(at: 1)
     }
     
     // MARK: DeleteRow Tapped
     @objc func deleteTapped() {
         moreButton.isHidden = true
         dropdownView.isHidden = true
-        if numberOfRows == 1 {
-            
-        } else {
-            updateRowNumber()
+        if numberOfRows != 1 {
             numberOfRows -= 1
+            updateRowNumber()
+            valueData.remove(at: (cellSelectedIndexPath?.section ?? 0) - 1)
```

Comment:
What's the different between valueData and tableCellsData below? Anyway to name these better or add comments to describe what's going on?

Reply:
I renamed valueData  to tableValue (as it contains the values of tableFiled) and tableCellsData contains the value of cells coming from joydoc.